lib/honeykiq/server_middleware.rb

```diff
@@ -3,20 +3,20 @@
 module Honeykiq
   class ServerMiddleware
     def initialize(options = {})
-      @honey_client = options.fetch(:honey_client)
+      @honey_client = options[:honey_client]
     end
 
     def call(_worker, msg, queue_name)
-      event = @honey_client.event
+      job = Sidekiq::Job.new(msg, queue_name)
 
-      call_event(event, msg, queue_name) { yield }
-    rescue StandardError => error
-      event&.add_field(:'job.status', 'failed')
-      event&.add(error_info(error))
-      raise
-    ensure
-      event&.add(extra_fields)
-      event&.send
+      start_span(name: job.display_class) do |event|
+        call_with_event(event, job, queue_name) { yield }
+      rescue StandardError => error
+        on_error(event, error)
+        raise
+      ensure
+        event&.add(extra_fields)
```

Comment:
Maybe this should be moved into `call_with_event`?
Feels like that will consolidate where we do the majority of the event manipulation